lib/pure/collections/deques.nim

```diff
@@ -84,6 +84,12 @@ proc initDeque*[T](initialSize: int = 4): Deque[T] =
   ## The length of a newly created deque will still be 0.
   result.initImpl(initialSize)
 
+proc initDeque*[T](s: seq[T]): Deque[T] =
+  ## Create a new deque initialized with a sequence
+  result = initDeque[T]()
```

Comment:
You should probably also use the initialSize argument for initDeque and set it to https://nim-lang.org/docs/math.html#nextPowerOfTwo,int of the length of the seq

Reply:
Oh wait, you could not call the original proc at all and just use result.initImpl(initialSize)

Reply:
Ok, i'm going to fix all that.

Reply:
Please, can you explain to me the use of initImpl ? It is not commented.

Reply:
@archnim just check how normal initDeque defined and use the same :)

Reply:
Line 25 of lib/pure/collections/deques.nim:
`var a = initDeque[int]()`

Isn't it correct ? Sorry i'm not yet a great professional of nim.

Reply:
@archnim See how initDeque is defined in https://github.com/nim-lang/Nim/blob/devel/lib/pure/collections/deques.nim#L85 .

Reply:
Thank you. It will be done in few minutes.